### PR TITLE
feature/8217-Chanel-Accessibility-checks-into-CI-documentation-site

### PR DESCRIPTION
**Pull request by IsraelleHub**
## Description of Change

Currently, we have CI processes that will run when the documentation site is updated. In looking at the deployed site, there seem to be some level of basic accessibility issues that we should be able to scan for. In this ticket we should create a point of automation that runs in a pull request, to check for accessibility issues.

This workflow run on PRs that touch the documentation site as well as merges into develop. If, on first run, there are failing accessibility issues reach out to [@timwright12](https://github.com/timwright12) to get them fixed or prioritized.  -->

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## Definition of Done

- [ ] An accessibility check is added into the CI process for documentation PRs
- [ ] If a failure happens while running in develop, alert slack in #va-mobile-build-alerts
- [ ] Checks run over all pages in the doc site
- [ ] Evidence of the workflow passing and failing (to confirm that it works)
- [ ] Workflow should be required (if possible)

## PR Checklist
  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: .github/workflows/documentation_accessibility_checks.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    needs: test-deploy
+    
+    steps:
+      - name: Notify Slack
+        env:
+          SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+          channel_name: va-mobile-build-alerts
+          message: 'On demand build started by ${{ github.actor }}: \"${{ inputs.notes }}\"  (:git: `${{ github.ref_name }}`). Accessibility issues detected. Please review and fix. See :thread: or <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|workflow run> for results. @Tim Wright'
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:99]: This message should be updated, it looks like it's just the on demand build one

File: .github/workflows/documentation_accessibility_checks.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  workflow_dispatch:
+    inputs:
+      notes:
+        description: Notes
+        required: true
+  # Run on Weekdays 9:00 UTC, 4AM ET, 1AM PT
+  schedule:
+    - cron: '0 9 * * 1,2,3,4,5'
+
+    # paths:
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:15]: Let's remove commented code if we're not using it

File: .github/workflows/documentation_accessibility_checks.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
+name: '[Documentation] Accessibility Check'
+
+# Adapted from https://docusaurus.io/docs/deployment
+
+on:
+  workflow_dispatch:
+    inputs:
+      notes:
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:8]: Probably don't need required notes on an accessibility scan

File: .github/workflows/documentation_accessibility_checks.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  schedule:
+    - cron: '0 9 * * 1,2,3,4,5'
+
+    # paths: 
+    #   - VAMobile/src/components/**
+    #   - VAMobile/documentation/**
+    #   - VAMobile/src/utils/hooks.tsx
+
+jobs:
+  test-deploy:
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:21]: Rename to something more accurate

File: .github/workflows/documentation_accessibility_checks.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        run: npm cache clean --force
+
+      # Step 2: Install Axe CLI globally
+      - name: Install Axe CLI globally
+        run: npm install -g @axe-core/cli@latest
+
+      # Step 3: Update npm
+      - name: Update npm
+        run: npm install -g npm@latest
+
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:37]: just clean up spacing

File: .github/workflows/documentation_accessibility_checks.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      - name: Start web server
+        working-directory: VAMobile/documentation
+        run: npm start & npx wait-on http://localhost:3000 &          
+      
+      # Step 7: Check accessibility issues and notify Accessibility Team
+      - name: Check accessibility issues
+        id: accessibility_check
+        run: |
+          # Fetch sitemap
+          wget https://department-of-veterans-affairs.github.io/va-mobile-app/sitemap.xml -O sitemap.xml
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:65]: Use local version of the site map and URLs

File: .github/workflows/documentation_test_build.yml [hunk 7]
```diff
@@ -28,3 +28,5 @@ jobs:
           cd documentation
           yarn install --frozen-lockfile
           yarn build
+
```
Comment (timwright12) [inline .github/workflows/documentation_test_build.yml:31]: clean up

**Review by timwright12 — CHANGES_REQUESTED** [review]

File: .github/workflows/documentation_accessibility_checks.yml [hunk 8]
```diff
@@ -0,0 +1,93 @@
+name: '[Documentation] Accessibility Check'
+
+# Adapted from https://docusaurus.io/docs/deployment
+
+on:
+  workflow_dispatch:
+  # Run on Weekdays 9:00 UTC, 4AM ET, 1AM PT
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:7]: The cron seems like it's just 9:00 UTC

File: VAMobile/documentation/docs/Flagship design library/Components/Buttons and Links/BackButton.mdx [hunk 9]
```diff
@@ -1,5 +1,5 @@
-import CodeBlock from '@theme/CodeBlock'
 import BackButtonString from '!!raw-loader!@componentsDocs/BackButton'
+import CodeBlock from '@theme/CodeBlock'
```
[the remark this answers is not in the document]
Comment (timwright12) [reply VAMobile/documentation/docs/Flagship design library/Components/Buttons and Links/BackButton.mdx:2]: is this part of the PR or in there by accident?

File: .github/workflows/documentation_accessibility_checks.yml [hunk 10]
```diff
@@ -0,0 +1,90 @@
+name: '[Documentation] Accessibility Check'
+
+# Adapted from https://docusaurus.io/docs/deployment
+
+on:
+  workflow_dispatch:
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:6]: Can we add a branch option here? There's an example of this in the on demand build workflow

File: .github/workflows/documentation_accessibility_checks.yml [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    needs: axe-scan
+    
+    steps:
+      - name: Notify Slack
+        env:
+          SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+          channel_name: va-mobile-build-alerts
+          message: 'Accessibility issues detected. Please review and fix. Build started by ${{ github.actor }}: (:git: `${{ github.ref_name }}`). See :thread: or <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|workflow run> for results.'
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:84]: Is there a way to output the amount of errors in the slack message?

File: .github/workflows/documentation_accessibility_checks.yml [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+      # Step 6: Start web server
+      - name: Start web server
+        working-directory: VAMobile/documentation
+        run: npm start & npx wait-on http://localhost:3000 &          
+      # Step 7: Check accessibility issues and notify Accessibility Team
+      - name: Check accessibility issues
+        id: accessibility_check
+        run: |
+          # Path to the downloaded sitemap file
+          sitemap_path="VAMobile/documentation/build/sitemap.xml"
```
Comment (timwright12) [inline .github/workflows/documentation_accessibility_checks.yml:49]: is this reporting local URLs or remote ones in the generated map?

File: .github/workflows/documentation_accessibility_checks.yml (hunk 10 above)
Comment (narin) [reply .github/workflows/documentation_accessibility_checks.yml:6]: I think we'll get this "for free" once the workflow has been merged to `develop` at least once. Additional comments on this below regarding the site map.

**Pull request closed without merging**